### ISSUE TITLE
system_images.ingressBackend in cluster.yml not used? 

### ISSUE THREAD

**Issue by carloscarnero** [rancher/rke#850]
**RKE version:**

```
rke --version
rke version v0.1.8
```

**Docker version: (`docker version`,`docker info` preferred)**

```
Containers: 19
 Running: 15
 Paused: 0
 Stopped: 4
Images: 9
Server Version: 1.13.1
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: journald
Cgroup Driver: systemd
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: docker-runc runc
Default Runtime: docker-runc
Init Binary: /usr/libexec/docker/docker-init-current
containerd version:  (expected: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1)
runc version: 5eda6f6fd0c2884c2c8e78a6e7119e8d0ecedb77 (expected: 9df8b306d01f59d3a8029be411de015b7304dd8f)
init version: fec3683b971d9c3ef73f284f176672c44b448662 (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Security Options:
 seccomp
  WARNING: You're not using the default seccomp profile
  Profile: /etc/docker/seccomp.json
 selinux
Kernel Version: 3.10.0-862.9.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 3
CPUs: 8
Total Memory: 5.827 GiB
Name: 307309d8.example.com
ID: BYO3:AQRU:UCN5:LWOY:TUGQ:HJF3:IWNN:5NLR:7C3N:DMTR:2X63:6LEQ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Registries: docker.io (secure)
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**

```
cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"
CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

root@307309d8 # uname -r
3.10.0-862.9.1.el7.x86_64
```

The system is a fully updated CentOS 7 (roughly equivalent to what RHEL 7.5 is).

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

Bare-metal.

**cluster.yml file:**

```
---
cluster_name: rancher
nodes:
- address: cfdd9f3c.example.com
  user: dockeruser
  role:
  - controlplane
  - etcd
  - worker
  ssh_key_path: ~/.ssh/id_rsa
- address: b5833011.example.com
  user: dockeruser
  role:
  - controlplane
  - etcd
  - worker
  ssh_key_path: ~/.ssh/id_rsa
- address: 307309d8.example.com
  user: dockeruser
  role:
  - controlplane
  - etcd
  - worker
  ssh_key_path: ~/.ssh/id_rsa
addons: |-
  ---
  kind: Namespace
  apiVersion: v1
  metadata:
    name: cattle-system
  ---
  kind: ServiceAccount
  apiVersion: v1
  metadata:
    name: cattle-admin
    namespace: cattle-system
  ---
  kind: ClusterRoleBinding
  apiVersion: rbac.authorization.k8s.io/v1
  metadata:
    name: cattle-crb
    namespace: cattle-system
  subjects:
  - kind: ServiceAccount
    name: cattle-admin
    namespace: cattle-system
  roleRef:
    kind: ClusterRole
    name: cluster-admin
    apiGroup: rbac.authorization.k8s.io
  ---
  apiVersion: v1
  kind: Service
  metadata:
    namespace: cattle-system
    name: cattle-service
    labels:
      app: cattle
  spec:
    ports:
    - port: 80
      targetPort: 80
      protocol: TCP
      name: http
    selector:
      app: cattle
  ---
  apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    namespace: cattle-system
    name: cattle-ingress-http
    annotations:
      nginx.ingress.kubernetes.io/proxy-connect-timeout: "30"
      nginx.ingress.kubernetes.io/proxy-read-timeout: "1800"
      nginx.ingress.kubernetes.io/proxy-send-timeout: "1800"
      nginx.ingress.kubernetes.io/ssl-redirect: "false"
  spec:
    rules:
    - host: rancher.example.com
      http:
        paths:
        - backend:
            serviceName: cattle-service
            servicePort: 80
  ---
  kind: Deployment
  apiVersion: extensions/v1beta1
  metadata:
    namespace: cattle-system
    name: cattle
  spec:
    replicas: 1
    template:
      metadata:
        labels:
          app: cattle
      spec:
        serviceAccountName: cattle-admin
        containers:
        - image: docker.example.com:5000/substratum/rancher:v2.0.4
          env:
          - name: CATTLE_AGENT_IMAGE
            value: "docker.example.com:5000/substratum/rancher-agent:v2.0.4"
          args:
          - --no-cacerts
          imagePullPolicy: Always
          name: cattle-server
          ports:
          - containerPort: 80
            protocol: TCP
system_images:
  kubernetes: docker.example.com:5000/substratum/hyperkube:v1.10.3-rancher2
  etcd: docker.example.com:5000/substratum/coreos-etcd:v3.1.12
  alpine: docker.example.com:5000/substratum/rke-tools:v0.1.10
  nginx_proxy: docker.example.com:5000/substratum/rke-tools:v0.1.10
  cert_downloader: docker.example.com:5000/substratum/rke-tools:v0.1.10
  kubernetes_services_sidecar: docker.example.com:5000/substratum/rke-tools:v0.1.10
  kubedns: docker.example.com:5000/substratum/k8s-dns-kube-dns-amd64:1.14.8
  dnsmasq: docker.example.com:5000/substratum/k8s-dns-dnsmasq-nanny-amd64:1.14.8
  kubedns_sidecar: docker.example.com:5000/substratum/k8s-dns-sidecar-amd64:1.14.8
  kubedns_autoscaler: docker.example.com:5000/substratum/cluster-proportional-autoscaler-amd64:1.0.0
  pod_infra_container: docker.example.com:5000/substratum/pause-amd64:3.1
  flannel: docker.example.com:5000/substratum/coreos-flannel:v0.9.1
  flannel_cni: docker.example.com:5000/substratum/coreos-flannel-cni:v0.2.0
  calico_node: docker.example.com:5000/substratum/calico-node:v3.1.1
  calico_cni: docker.example.com:5000/substratum/calico-cni:v3.1.1
  calico_ctl: docker.example.com:5000/substratum/calico-ctl:v2.0.0
  canal_node: docker.example.com:5000/substratum/calico-node:v3.1.1
  canal_cni: docker.example.com:5000/substratum/calico-cni:v3.1.1
  canal_flannel: docker.example.com:5000/substratum/coreos-flannel:v0.9.1
  weave_node: weaveworks/weave-kube:2.1.2
  weave_cni: weaveworks/weave-npc:2.1.2
  ingress: docker.example.com:5000/substratum/nginx-ingress-controller:0.10.2-rancher3
  ingressBackend: docker.example.com:5000/substratum/nginx-ingress-controller-defaultbackend:1.4
```

**Steps to Reproduce:**

I'm trying to set up a HA cluster for Rancher with an external load balancer [as documented](https://rancher.com/docs/rancher/v2.x/en/installation/ha-server-install-external-lb/). This cluster in running on-premises, in an air-gapped environment; that's why the system images live in `docker.example.com` registry.

**Results:**

After running `rke up --config cluster.yml` everything gets installed perfectly (Rancher, its UI, etc.) except the default backend por for Nginx:

```
kubectl get pods --all-namespaces
NAMESPACE       NAME                                      READY     STATUS             RESTARTS   AGE
cattle-system   cattle-698bfb86b9-9xq2q                   1/1       Running            0          2h
cattle-system   cattle-cluster-agent-79bdd94bf8-f7jjh     1/1       Running            0          2h
cattle-system   cattle-node-agent-8vj6f                   1/1       Running            0          2h
cattle-system   cattle-node-agent-9mhpw                   1/1       Running            0          2h
cattle-system   cattle-node-agent-s8gnh                   1/1       Running            0          2h
ingress-nginx   default-http-backend-564b9b6c5b-clp69     0/1       ImagePullBackOff   0          2h
ingress-nginx   nginx-ingress-controller-8g55g            1/1       Running            0          2h
ingress-nginx   nginx-ingress-controller-c7jq6            1/1       Running            0          2h
ingress-nginx   nginx-ingress-controller-r8c4k            1/1       Running            0          2h
kube-system     canal-5x8j6                               3/3       Running            0          2h
kube-system     canal-6wxz9                               3/3       Running            1          2h
kube-system     canal-ndftb                               3/3       Running            0          2h
kube-system     kube-dns-69c948c598-7spmr                 3/3       Running            0          2h
kube-system     kube-dns-autoscaler-79c8f7cbcd-6tfj2      1/1       Running            0          2h
kube-system     rke-ingress-controller-deploy-job-6r8ql   0/1       Completed          0          2h
kube-system     rke-kubedns-addon-deploy-job-45dqs        0/1       Completed          0          2h
kube-system     rke-network-plugin-deploy-job-tmgjv       0/1       Completed          0          2h
kube-system     rke-user-addon-deploy-job-ks9tv           0/1       Completed          0          2h
```

Note above the status of the pod `default-http-backend-564b9b6c5b-clp69`. Checking the logs I find several messages like:

```
ago 08 14:35:28 307309d8.example.com dockerd-current[1043]: E0808 18:35:28.344438    3959 pod_workers.go:186] Error syncing pod 3dc2b6e4-9b21-11e8-9d89-000c29215f29 ("default-http-backend-564b9b6c5b-clp69_ingress-nginx(3dc2b6e4-9b21-11e8-9d89-000c29215f29)"), skipping: failed to "StartContainer" for "default-http-backend" with ImagePullBackOff: "Back-off pulling image \"rancher/nginx-ingress-controller-defaultbackend:1.4\""
```

Note that it seems to be trying to pull `rancher/nginx-ingress-controller-defaultbackend:1.4` instead of `docker.example.com:5000/substratum/nginx-ingress-controller-defaultbackend:1.4` as specified in the `cluster.yml` file above.

Allow me to repeat that every other image gets pulled from the internal registry as expected, including the actual ingress controller. It's just the default backend that is not using the image defined.

**Comment by sebastien-prudhomme** [comment]
Try "ingress_backend:" instead of "ingressBackend:" in your YAML file, it should work better.

**Comment by carloscarnero** [comment]
Yes! It worked as expected. IMO, this means that there's a minor error [in the documentation](https://github.com/rancher/docs/blob/master/content/rke/v0.1.x/en/config-options/system-images/_index.md).

**Comment by carloscarnero** [comment]
Since it is obviously a little problem with the documentation, I'll close this issue. I did open one in the docs project, rancher/docs#591 with a fix request, rancher/docs#592.
